extensions-core/avro-extensions/src/main/java/org/apache/druid/data/input/avro/SchemaRegistryBasedAvroBytesDecoder.java

```diff
@@ -91,17 +98,62 @@ public int getCapacity()
   }
 
   @JsonProperty
-  public Map<String, ?> getConfig()
+  public Map<String, Object> getConfig()
   {
     return config;
   }
 
   @JsonProperty
-  public Map<String, String> getHeaders()
+  public Map<String, Object> getHeaders()
   {
     return headers;
   }
 
+  protected Map<String, String> createRegistryHeader()
```

Comment:
These methods all look basically the same for both avro and protobuf (and kafka).

I suggest making a static method or two, perhaps in a new class, maybe `DynamicConfigProviderUtils` or something similar in `druid-core`, that takes a `Map<String, Object>` and performs this logic

Reply:
Thanks for reviewing! I add a DynamicConfigProviderUtils which provides function like ```createRegistryHeader()```, but it can not adapt to kafka.

Reply:
oops, it looks like my suggestion made coverage check fail because the tests that cover `DynamicConfigProviderUtils` live in a different module. If it isn't too much trouble could add a test for it and it should make it go away.

I think its ok to do the changes to kafka in a follow-up if you would prefer.

Reply:
I will add unit tests for it. As for Kafka, In [#11372](https://github.com/apache/druid/pull/11372), I add DynamicConfigProvider for Kafka in producer properties, I think I can do the changes for Kafka in this [#11372](https://github.com/apache/druid/pull/11372), do you think it is suitable?

Reply:
Unit tests added.